src/components/implementation/no_interface/llbooter/boot_deps.h

```diff
@@ -105,6 +139,96 @@ boot_newcomp_sinv_alloc(spdid_t spdid)
 	}
 }
 
+/* TODO: possible to cos_defcompinfo_child_alloc if we somehow move allocations to one place */
+static void
+boot_newcomp_defcinfo_init(spdid_t spdid)
+{
+	struct cos_defcompinfo *defci     = cos_defcompinfo_curr_get();
+	struct cos_aep_info *   sched_aep = cos_sched_aep_get(defci);
+	struct cos_aep_info *   child_aep = cos_sched_aep_get(new_comp_cap_info[spdid].defci);
+	struct cos_compinfo *   child_ci  = cos_compinfo_get(new_comp_cap_info[spdid].defci);
+
+	child_aep->thd = cos_initthd_alloc(boot_info, child_ci->comp_cap);
+	assert(child_aep->thd);
+
+	if (new_comp_cap_info[spdid].is_sched) {
```

Comment:
All of this is going to conflict with @RobertGiff 's PRs that he is wrapping up now.  You two need to talk.

Reply:
@RobertGiff, lets talk about our PRs. Monday in lab? I'll ping you on slack anyway!